core/src/main/scala/org/apache/spark/sql/internal/JarUtils.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.sql.internal
+
+import java.io.File
+import java.net.{URL, URLClassLoader}
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.scheduler.local.LocalSchedulerBackend
+import org.apache.spark.sql.collection.{ToolsCallbackInit}
+import org.apache.spark.sql.{SnappyContext, SnappySession}
+import org.apache.spark.util.{Utils, MutableURLClassLoader}
+import org.apache.spark.{SparkEnv, SparkFiles, Logging, SparkContext}
+
+/**
+ * An utility class to store jar file reference with their individual classloaders. This is reflect class changes at driver side.
+ * e.g. If an UDF definition changes the driver should pick up the correct UDF class.
+ * This class can not initialize itself after a driver failure. So the callers will have  to make sure that the classloader gets
+ * initialized after a driver startup. Usually it can be achieved by adding classloader at query time.
+ *
+ */
+class ContextJarUtils(sparkContext: SparkContext) extends Logging{
+
+  val workingDir = new File(System.getProperty("user.dir"))
```

Comment:
I have modified the code to add a separate folder snappy-jars inside user directory.